test/e2e_node/eviction_test.go

```diff
@@ -0,0 +1,567 @@
+		// This means the eviction manager will observe pressure before we will, creating a delay between when the eviction manager
+		// evicts a pod, and when we observe the pressure by querrying the API server.  Add a delay here to account for this delay
+		By("making sure pressure from test has surfaced before continuing")
+		time.Sleep(pressureDelay)
+
+		By("making sure conditions eventually return to normal")
+		Eventually(func() error {
+			logFunc()
+			if pressure != nil && hasPressure(f, pressure) {
+				return fmt.Errorf("Conditions havent returned to normal, we still have %s", testCondition)
+			}
+			return nil
+		}, pressureDissapearTimeout, evictionPollInterval).Should(BeNil())
+
+		By("making sure conditions do not return, and that pods that shouldnt fail dont fail")
+		Consistently(func() error {
+			logFunc()
+			if pressure != nil && hasPressure(f, pressure) {
+				return fmt.Errorf("%s dissappeared and then reappeared", testCondition)
+			}
+			// Gather current information
+			updatedPodList, _ := f.ClientSet.Core().Pods(f.Namespace.Name).List(metav1.ListOptions{})
+			for _, priorityPodSpec := range podTestSpecs {
+				// EvictionPriority 0 pods should not fail
+				if priorityPodSpec.evictionPriority == 0 {
+					for _, p := range updatedPodList.Items {
+						if p.Name == priorityPodSpec.pod.Name && p.Status.Phase == v1.PodFailed {
+							logKubeletMetrics(kubeletmetrics.EvictionStatsAgeKey)
+							return fmt.Errorf("%s pod failed (delayed) and shouldn't have failed", p.Name)
+						}
+					}
+				}
+			}
+			return nil
+		}, postTestConditionMonitoringPeriod, evictionPollInterval).Should(BeNil())
+
+		By("making sure we can start a new pod after the test")
+		podName := "test-admit-pod"
+		f.PodClient().CreateSync(&v1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: podName,
+			},
+			Spec: v1.PodSpec{
+				RestartPolicy: v1.RestartPolicyNever,
+				Containers: []v1.Container{
+					{
+						Image: framework.GetPauseImageNameForHostArch(),
+						Name:  podName,
+					},
+				},
+			},
+		})
+	})
+
+	AfterEach(func() {
+		By("deleting pods")
+		for _, spec := range podTestSpecs {
+			By(fmt.Sprintf("deleting pod: %s", spec.pod.Name))
+			f.PodClient().DeleteSync(spec.pod.Name, &metav1.DeleteOptions{}, longPodDeletionTimeout)
+		}
+
+		if CurrentGinkgoTestDescription().Failed {
+			if framework.TestContext.DumpLogsOnFailure {
+				logPodEvents(f)
+				logNodeEvents(f)
+			}
+			By("sleeping to allow for cleanup of test")
+			time.Sleep(postTestConditionMonitoringPeriod)
+		}
+	})
+}
+
+// Returns TRUE if the node has pressure, FALSE otherwise
+func hasPressure(f *framework.Framework, nodeCondition *v1.NodeConditionType) bool {
+	localNodeStatus := getLocalNode(f).Status
+	_, pressure := nodeutil.GetNodeCondition(&localNodeStatus, *nodeCondition)
+	Expect(pressure).NotTo(BeNil())
+	hasPressure := pressure.Status == v1.ConditionTrue
+	By(fmt.Sprintf("checking if pod has %v: %v", pressure, hasPressure))
```

Comment:
I don't like coupling output to a predicate.

Reply:
not sure what you mean, but I changed this to give different messages based on hasPressure

Reply:
Side-effects (e.g. logging) from boolean predicates are unexpected. See my other comment on the `hasNodeCondition` function.